lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/TransactionSpec.hs

```diff
@@ -0,0 +1,173 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.Jormungandr.TransactionSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Jormungandr.Compatibility
+    ( Jormungandr )
+import Cardano.Wallet.Jormungandr.Environment
+    ( KnownNetwork (..), Network (..) )
+import Cardano.Wallet.Jormungandr.Transaction
+    ( newTransactionLayer )
+import Cardano.Wallet.Primitive.CoinSelection
+    ( CoinSelection (..) )
+import Cardano.Wallet.Primitive.CoinSelection.LargestFirst
+    ( largestFirst )
+import Cardano.Wallet.Primitive.Types
+    ( Address (..)
+    , Coin (..)
+    , Hash (..)
+    , ShowFmt (..)
+    , TxIn (..)
+    , TxOut (..)
+    , UTxO (..)
+    )
+import Cardano.Wallet.Transaction
+    ( TransactionLayer (..) )
+import Control.Monad.Trans.Except
+    ( runExceptT )
+import Data.Functor.Identity
+    ( Identity (runIdentity) )
+import Data.List.NonEmpty
+    ( NonEmpty )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Quantity
+    ( Quantity (..) )
+import Test.Hspec
+    ( Spec, describe, it )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Gen
+    , Property
+    , choose
+    , property
+    , scale
+    , vectorOf
+    , withMaxSuccess
+    , (===)
+    )
+
+import qualified Cardano.Wallet.Primitive.CoinSelection as CS
+import qualified Data.ByteString as BS
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Map as Map
+
+spec :: Spec
+spec = do
+    describe "estimateSize" $ do
+        it "Estimated size is zero"
+            (withMaxSuccess 2500 $ property $ propSizeEstimation $ Proxy @'Mainnet)
+        it "Estimated size is zero"
+            (withMaxSuccess 2500 $ property $ propSizeEstimation $ Proxy @'Testnet)
+
+{-------------------------------------------------------------------------------
+                                Size Estimation
+-------------------------------------------------------------------------------}
+
+propSizeEstimation
+    :: forall n. (KnownNetwork n)
+    => Proxy n
+    -> (ShowFmt CoinSelection)
+    -> Property
+propSizeEstimation _ (ShowFmt sel) =
+    let
+        blockHash =
+            Hash {getHash = "\216OY\rX\199\234\188.<O\\\244Y\211\210\254\224`i\216\DC3\167\132\139\154\216\161T\174\247\155"}
+        tl = newTransactionLayer blockHash :: TransactionLayer (Jormungandr n)
+        calcSize = estimateSize tl sel
+    in calcSize === Quantity 0
```

Comment:
So I take it that we're going to implement the size calculation the transaction layer after all? Or are you planning to leave the quantity at 0? In which case, the whole property + arbitrary instances is maybe overkill don't you think :sweat_smile:  ?

Reply:
yeah, coin selection **at this moment** should probably be as dummy as possible. But it will change soon (that is my understanding) and Jormungandr will support fee calculation and they will be valid...

Reply:
Generally speaking, it's best to not do things that aren't required on the moment. We have actually no idea when Jormungandr will support fee calculation, although we know that they eventually will. 

But, now that it's done, I'd almost be tempted to really implement the size estimator...